File: src/main/java/pt/joaomneto/titancompanion/adventurecreation/impl/SOTSAdventureCreation.java
package pt.joaomneto.titancompanion.adventurecreation.impl;

import java.io.BufferedWriter;
import java.io.IOException;

import pt.joaomneto.titancompanion.R;
import pt.joaomneto.titancompanion.adventure.Adventure.AdventureFragmentRunner;
import pt.joaomneto.titancompanion.adventurecreation.AdventureCreation;
import pt.joaomneto.titancompanion.adventurecreation.impl.fragments.sots.SOTSMartialArt;

import android.view.View;

public class SOTSAdventureCreation extends AdventureCreation {

	SOTSMartialArt skill;

	public SOTSAdventureCreation() {
		super();
		fragmentConfiguration.clear();
		fragmentConfiguration.put(0, new AdventureFragmentRunner(R.string.title_adventure_creation_vitalstats, "pt.joaomneto.titancompanion.adventurecreation.impl.fragments.VitalStatisticsFragment"));
		fragmentConfiguration.put(1, new AdventureFragmentRunner(R.string.title_adventure_creation_skill,
				"pt.joaomneto.titancompanion.adventurecreation.impl.fragments.sots.SOTSAdventureCreationSkillFragment"));

	}

	@Override
	protected void rollGamebookSpecificStats(View view) {

	}



	@Override
	protected void storeAdventureSpecificValuesInFile(BufferedWriter bw) throws IOException {
		bw.write("honour=3\n");
		bw.write("skill=" + skill.name() + "\n");
		bw.write("provisions=10\n");
		bw.write("provisionsValue=4\n");
		bw.write("gold=0\n");
		if (SOTSMartialArt.KYUJUTSU == skill) {
			bw.write("willowLeafArrows=3\n");
			bw.write("bowelRakerArrows=3\n");
			bw.write("armourPiercerArrows=3\n");
			bw.write("hummingBulbArrows=3\n");
		}
	}


	public SOTSMartialArt getSkill() {
		return skill;
	}

	public void setSkill(SOTSMartialArt skill) {
		this.skill = skill;
	}


	@Override
	public String validateCreationSpecificParameters() {
		StringBuilder sb = new StringBuilder();
		if(this.skill == null){
			sb.append("Martial Art");
		}
		return  sb.toString();
	}

}

File: src/main/java/pt/joaomneto/titancompanion/adventurecreation/impl/SOTSAdventureCreation.kt
package pt.joaomneto.titancompanion.adventurecreation.impl

import android.view.View
import pt.joaomneto.titancompanion.R
import pt.joaomneto.titancompanion.adventurecreation.AdventureCreation
import pt.joaomneto.titancompanion.adventurecreation.impl.fragments.VitalStatisticsFragment
import pt.joaomneto.titancompanion.adventurecreation.impl.fragments.sots.SOTSAdventureCreationSkillFragment
import pt.joaomneto.titancompanion.adventurecreation.impl.fragments.sots.SOTSMartialArt
import pt.joaomneto.titancompanion.util.AdventureFragmentRunner
import java.io.BufferedWriter
import java.io.IOException

class SOTSAdventureCreation : AdventureCreation(
    arrayOf(
        AdventureFragmentRunner(R.string.title_adventure_creation_vitalstats, VitalStatisticsFragment::class),
        AdventureFragmentRunner(
            R.string.title_adventure_creation_skill,
            SOTSAdventureCreationSkillFragment::class
        )
    )
) {

    private var martialArt: SOTSMartialArt? = null

    override fun rollGamebookSpecificStats(view: View) {}

    @Throws(IOException::class)
    override fun storeAdventureSpecificValuesInFile(bw: BufferedWriter) {
        bw.write("honour=3\n")
        bw.write("skill=" + martialArt!!.name + "\n")
        bw.write("provisions=10\n")
        bw.write("provisionsValue=4\n")
        bw.write("gold=0\n")
        if (SOTSMartialArt.KYUJUTSU == martialArt) {
            bw.write("willowLeafArrows=3\n")
            bw.write("bowelRakerArrows=3\n")
            bw.write("armourPiercerArrows=3\n")
            bw.write("hummingBulbArrows=3\n")
        }
    }

    fun setSkill(martialArt: SOTSMartialArt) {
        this.martialArt = martialArt
    }

    override fun validateCreationSpecificParameters(): String? {
        val sb = StringBuilder()
        if (this.martialArt == null) {
            sb.append("Martial Art")
        }
        return sb.toString()
    }
}
